x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportGetDataFrameTransformsStatsAction.java

```diff
@@ -224,6 +242,37 @@ private void collectStatsForTransformsWithoutTasks(Request request,
             searchStatsListener, client::search);
     }
 
+    private void getCheckpointingStateAndStatsWithoutTask(ClusterState state, DataFrameIndexerTransformStats stats,
```

Comment:
This method only captures the checkpoint stats in the rare case that the task is still in cluster state, but NOT allocated to a node. Which, I think, is not really possible with how transforms is written right now, unless persistent tasks are turned off on all the nodes.